origin-js/package.json

```diff
@@ -42,13 +42,18 @@
     "crypto-random-string": "^1.0.0",
     "ethereumjs-util": "^5.2.0",
     "form-data": "^2.3.2",
+    "fs-extra": "^7.0.0",
```

Comment:
Yea that is a tricky one. What I'm trying to do here is avoid installing all the dev dependencies because it greatly speeds up the builds. But I need to install the dependencies that are required for building production. I might be able to refactor to remove fs-extra, and then just add npm install for truffle and webpack.

I'm not sure that those extra packages will alter the build size at all anyway. I think webpack won't include anything extra in the bundle that is not required by the source. So I don't think there is any disadvantage. I will test out the build sizes though.

Reply:
Got it. Thanks for the details.

Yes, you are right, I don't think the build size will be impacted by adding those dependencies. But when developers in the future will install the origin-js package, it will pull those unnecessary dependencies which is not great. See https://www.npmjs.com/package/origin

Hmm.... that's tricky. One idea could be to have a base container that contains node and all the tools/npm packages we need to build. And use that base container rather than "FROM node:9". A drawback of this approach is that, assuming we still want folks to have those tools installed locally when they develop on their host, then they would also need to be in the package.json of origin-js and other packages... and it may become difficult to keep in sync the definition of that base container with the package.json from those various packages ?

Reply:
Looking into this further I realised that these extra dependencies will get unnecessarily pulled when origin-js is installed, but since origin-js is an in browser bundle compiled by webpack none of the dependencies in the production dependency list need to be installed. If a developer runs `NODE_ENV=production npm install origin-js` ~1500 packages will be installed that aren't needed at all. So I think there is a larger problem here.

If you checkout the [package.json](https://github.com/vuejs/vue/blob/dev/package.json) for Vue you can see that everything is listed under devDependencies for this reason.

I'm not sure how to solve this but I suggest we deal with this as a separate issue.

Reply:
Great point regarding no need to pull those dependencies since we use webpack to bundle...
Agreed we can handle as a separate issue from this CI work. Do you want to file an issue to keep track of this ? I'll approve this CL in the meantime...

Reply:
Thanks, raised an issue for this!